packages/request-node/src/request/persistTransaction.ts

```diff
@@ -30,9 +30,22 @@ export default class PersistTransactionHandler {
     // Retrieves data access layer
     let dataAccessResponse: DataAccessTypes.IReturnPersistTransaction;
 
+    const transactionHash: MultiFormatTypes.HashTypes.IHash = normalizeKeccak256Hash(
+      clientRequest.body.transactionData,
+    );
+
     // Set the timeout from the value from config and convert seconds to milliseconds
     /* eslint-disable no-magic-numbers */
-    clientRequest.setTimeout(getPersistTransactionTimeout() * 1000);
+    clientRequest.setTimeout(getPersistTransactionTimeout() * 1000, () => {
+      this.logger.error(
+        `persistTransaction timeout ${JSON.stringify({
+          transactionHash,
+          channelId: clientRequest.body.channelId,
+        })}`,
+        ['timeout'],
+      );
+      serverResponse.status(StatusCodes.GATEWAY_TIMEOUT).send('persistTransaction timeout');
+    });
```

Comment:
Here, again, I think we must validate that clientRequest.body.channelId exists before we log it in this statement.  Possibly with an if/else as is done in ipfsAdd or using a try/catch.